lib/manager/pip_requirements/extract.ts

```diff
@@ -13,16 +17,15 @@ const specifierPartPattern = `\\s*${rangePattern.replace(/\?<\w+>/g, '?:')}`;
 const specifierPattern = `${specifierPartPattern}(?:\\s*,${specifierPartPattern})*`;
 export const dependencyPattern = `(${packagePattern})(${extrasPattern})(${specifierPattern})`;
 
-export function extractPackageFile(
-  content: string,
-  _: string,
-  config: ExtractConfig
-): PackageFile | null {
+export function extractPackageFile({
+  fileContent,
+  config,
```

Comment:
Please change Line 40/43 from `if (config.registryUrls && config.registryUrls.length)` to `if (config?.registryUrls?.length)` as config is optional.

@rarkins  Or should config be required in `ExtractPackageFileConfig`?

Tests can now use mock<ExtractPackageFileConfig>({ <partial>}) from `jest-mock-extended` for getting a complete config.

Reply:
![Screenshot 2020-01-16 at 5 44 43 PM](https://user-images.githubusercontent.com/9268592/72524425-ed3a9000-3887-11ea-83a2-b14021a4b7bd.png)

Reply:
🙈 prettier can't read ts 3.7 code until 1.19

https://prettier.io/blog/2019/11/09/1.19.0.html

@rarkins how we should resovle this? update prettier or just do not use new syntactic sugar?

Reply:
I think there's been enough wait time to update to prettier 1.19 and make the necessary updates

Reply:
Back to the original question: we can make config required, because we'll always pass it as a param from the workers.

Reply:
Prettier 1.19 update applied, but I prefer to go with mandatory config and update tests if necessary

Reply:
Then we would write `if ( config.registryUrls?.length)`

Reply:
Changing in `interface` will cause a lot of type error. Is that okay?
Where config is not present, we will pass `{}` there.

Reply:
The type errors should only occur in tests. There you can use the mentioned `mock` function, which will mock missing properties and let the type check pass.